### ISSUE TITLE
Improve `openshift_transport.go` error messages

### ISSUE THREAD

**Issue by mtrmac** [containers/image#138]
In

``` go
    tagged, ok := r.(reference.NamedTagged)
    if !ok {
        return nil, fmt.Errorf("invalid image reference %s, %#v", ref, r)
    }
```

and

``` go
    r := strings.SplitN(dockerRef.RemoteName(), "/", 3)
    if len(r) != 2 {
        return nil, fmt.Errorf("invalid image reference %s", dockerRef.String())
    }
```

we know fairly precisely what we want and what we aren’t getting, so we should tell the users what exactly they need to do to have the input accepted, instead of vaguely rejecting the input as “invalid”.

**Comment by GrantSeltzer** [comment]
This can be closed: https://github.com/containers/image/pull/186

**Comment by mtrmac** [comment]
I don’t think so; we use `errors.Errorf`, which gives us a stack trace, but the original error values are not attached (these lines have been updated to use `errors.Errorf`, but not `errors.Wrap`)

**Comment by mtrmac** [comment]
… oops, there is no “original error” in these cases. So, yes, the stack trace is in principle sufficient to tell what is going on—for the developers. A readable error message would still be an improvement.

**Comment by runcom** [comment]
I'm ok adding more context to the error (it was my fault at writing those subtle errors). I'll fix this tomorrow.

**Comment by GrantSeltzer** [comment]
@runcom I can handle it, fixing it up now

**Comment by runcom** [comment]
Cool, thanks Grant!

**Comment by mtrmac** [comment]
This was fixed back in January by #209.
